DQM/SiPixelPhase1RecHits/src/SiPixelPhase1RecHits.cc

```diff
@@ -41,8 +42,16 @@ void SiPixelPhase1RecHits::analyze(const edm::Event& iEvent, const edm::EventSet
   iEvent.getByToken( srcToken_, tracks);
   if (!tracks.isValid()) return;
 
+  edm::Handle<reco::VertexCollection> vertices;
```

Comment:
hi @fioriNTU - please do the getByToken (and the consumes above) only if applyVertexCut_ is true
Also replace vertices->size()==0 by vertices->empty()